packages/contracts/contracts/consensus/BurnAuction.sol

```diff
@@ -136,8 +139,24 @@ contract BurnAuction is IConsensusProvider, IBurnAuction {
     /**
      * @notice This function will be updated as the governance of Zkopru's been updated.
      */
-    function isProposable(address proposer) public override view returns (bool) {
-        return activeCoordinator() == address(0) || activeCoordinator() == proposer;
+    function isProposable(address proposer) public override returns (bool) {
+        if (activeCoordinator() == address(0) || activeCoordinator() == proposer) {
+            return true;
+        }
+        if (latestOpenRound == currentRound()) return true;
+        uint currentRoundStart = calcRoundStart(currentRound());
+        if (block.timestamp > currentRoundStart + roundLength / 2) {
+            // If more than midway through the round determine if a block has
+            // been proposed. If not, open the round for anyone to propose blocks
+            uint latestProposalBlock =
+              ICoordinatable(address(zkopru)).coordinatorExitBlock(activeCoordinator()) - zkopru.CHALLENGE_PERIOD();
+            // approx block start
+            uint roundStartBlock = block.number - ((block.timestamp - currentRoundStart) / 15);
+            if (latestProposalBlock < roundStartBlock) {
+                latestOpenRound = currentRound();
+                return true;
+            }
+        }
```

Comment:
I think updating the `latestOpenRound` should be a separate function and it'll be good to keep this `isProposable` as a `view` function :)
How about adding `updateLatestOpenRound()` and call this function in `propose()` before we call `isProposable()`?

Reply:
Good call, I was kinda unhappy with this implementation.